pkg/cmd/addon.go

```diff
@@ -1,88 +1,108 @@
 package cmd
 
 import (
-	"context"
-	"encoding/json"
+	"errors"
 	"fmt"
 	"io/ioutil"
-	"net/http"
-	"os"
 	"path/filepath"
-	"strings"
+
+	"github.com/cloud-native-application/rudrx/pkg/utils/system"
+
+	"github.com/cloud-native-application/rudrx/pkg/plugins"
 
 	"github.com/cloud-native-application/rudrx/api/types"
-	corev1alpha2 "github.com/crossplane/oam-kubernetes-runtime/apis/core/v1alpha2"
-	"github.com/ghodss/yaml"
 
 	cmdutil "github.com/cloud-native-application/rudrx/pkg/cmd/util"
 	"github.com/gosuri/uitable"
 	"github.com/spf13/cobra"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var (
-	addonCenterConfigFile = ".vela/addon_config"
-	defaultAddonCenter    = "local"
-)
-
-//Used to store addon center config in file
-type AddonCenterConfig struct {
-	Name    string `json:"name"`
-	IsLocal bool   `json:"isLocal"`
-}
-type PluginFile struct {
-	Name string `json:"name"`
-	Url  string `json:"download_url"`
-	Sha  string `json:"sha"`
-}
-
-type Plugin struct {
-	Name       string `json:"name"`
-	Type       string `json:"type"`
-	Definition string `json:"definition"`
-	Status     string `json:"status"`
-	ApplesTo   string `json:"applies_to"`
-}
-
-func AddonCommandGroup(parentCmd *cobra.Command, c types.Args, ioStream cmdutil.IOStreams) {
-	parentCmd.AddCommand(NewAddonConfigCommand(ioStream),
-		NewAddonListCommand(c, ioStream),
+func AddonCommandGroup(parentCmd *cobra.Command, ioStream cmdutil.IOStreams) {
+	parentCmd.AddCommand(
+		NewAddonConfigCommand(ioStream),
+		NewAddonListCommand(ioStream),
+		NewAddonUpdateCommand(ioStream),
 	)
 }
 
 func NewAddonConfigCommand(ioStreams cmdutil.IOStreams) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "addon:config",
+		Use:     "addon:config <reponame> <url>",
```

Comment:
If we name every repository url, then we might need to `addon:switch` to switch the currently using repository. The name won't do more help to identify the url as the url itself would.

Reply:
we can list them all, for example:

```
NAME          STATUS
repo1/route   installed
repo2/scaler uninstalled
```

```
vela addon:install repo2/scaler
```